content/chainguard/chainguard-enforce/chainguard-enforce-kubernetes/gulfstream-overview/index.md

```diff
@@ -0,0 +1,90 @@
+
+
+## Background
+
+Gulfstream was born out of two rising trends within the Kubernetes ecosystem that, when combined, present a high level of complexity for cluster operators.
+
+The first is the sheer proliferation of clusters. It's increasingly common for organizations to have multiple clusters per Availability Zone (AZ), as in the following diagram.
+
+![Diagram showing multiple clusters per area zone. Three AZs are represented: us-west2 with Clusters A, B, and C; us-central3 with Clusters E and D; us-east1 with Clusters F, G, H, and I.](clusters_per_AZ.png)
+
+This trend is driven by a number of factors, including the following.
+
+* A desire to further reduce the blast radius of a cluster-wide outage by segmenting rollouts within an AZ further.
+* A tenancy model where each logical application gets its own cluster per AZ.
+* A need for more resources per AZ than Kubernetes currently supports in a single cluster.
+
+The second trend is the growth of extensions based on [Custom Resource Definitions](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) (CRDs) within Kubernetes. These kinds of extensions have been used for a variety of purposes, from certificate provisioning to higher level app abstractions to service mesh, as in the following diagram. The following diagram represents a cluster running a variety of common extensions.
+
+![Diagram representing a single cluster running the following extensions: cert-manager, Istio, Tekton, Knative, Kafka Operator, Kpack, Flux, Prometheus Operator.](single_cluster.png)
+
+CRDs are only made up of data — which is often schematized — but they are brought to life by controller processes and webhooks that conventionally run on the cluster itself, serving as a sort of extension to the built-in control plane processes. 
+
+With traditional controller architectures, every cluster in a fleet runs every one of the clusters needed to run certain abstractions. This means that it's increasingly common for platform operations teams to be managing multiple extensions across many clusters, as in this diagram.
+
+![Diagram similar to the first in this guide, showing multiple clusters across three area zones (us-west2, us-central3, and us-east1). However, this diagram highlights the extensions running on each cluster.](clusters_per_AZ_extensions.png)
+
+As the number of clusters grows, so does the difficulty and cost of offering managed extensions to each of them, potentially growing into a complexity nightmare.
+
+
+## Understanding Controllers, Reconcilers, and Control Planes
+
+With the spread of CRDs, it can be difficult to know how trustworthy a given extension is. When you extend Kubernetes with extensions based on CRDs, you install additional control planes which implement the semantics of their extension. A *control plane* consists of the various APIs that control the behavior of systems; in Kubernetes, the API server and its controllers form the control plane. 
```

Comment:
This is a tiny nit, but what about subbing "extend" for something else since "extensions" is so nearby? I know they are technical terms so maybe disregard. One possibility I thought of:

"When you broaden the scope of Kubernetes with extensions..."

Reply:
hadn't even noticed that, great suggestion!